auth/user_mgt_test.go

```diff
@@ -0,0 +1,716 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"firebase.google.com/go/internal"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type mockAuthServer struct {
+	Resp   []byte
+	Header map[string]string
+	Status int
+	Req    []*http.Request
+	rbody  []byte
+	srv    *httptest.Server
+	Client *Client
+}
+
+var listUsers []*ExportedUserRecord
+
+func TestGetUser(t *testing.T) {
+	s, closer := echoServer("get_user.json", t)
+	defer closer()
+
+	user, err := s.Client.GetUser(context.Background(), "ignored_id")
+	if err != nil {
+		t.Error(err)
+	}
+	want := &UserRecord{
+		UserInfo: &UserInfo{
+			UID:         "testuser",
+			Email:       "testuser@example.com",
+			PhoneNumber: "+1234567890",
+			DisplayName: "Test User",
+			PhotoURL:    "http://www.example.com/testuser/photo.png",
+		},
+		Disabled: false,
+
+		EmailVerified: true,
+		ProviderUserInfo: []*UserInfo{
+			{
+				ProviderID:  "password",
+				DisplayName: "Test User",
+				PhotoURL:    "http://www.example.com/testuser/photo.png",
+				Email:       "testuser@example.com",
+			}, {
+				ProviderID:  "phone",
+				PhoneNumber: "+1234567890",
+			},
+		},
+		UserMetadata: &UserMetadata{
+			CreationTimestamp:  1234567890,
+			LastLogInTimestamp: 1233211232,
+		},
+		CustomClaims: map[string]interface{}{"admin": true, "package": "gold"},
+	}
+	if !reflect.DeepEqual(user, want) {
+		t.Errorf("got %#v, wanted %#v", user, want)
```

Comment:
This still doesn't quite follow the convention followed in the rest of the codebase. I would expect to see the function name in the error message. For example:

```
t.Errorf("GetUser() = %v; want = %v", user, want)
```

This should be followed in pretty much everywhere in this file.